### PR TITLE
Scanner: Filter out not applicable license files for packages

### PR DESCRIPTION
**Pull request by fviernau**
The filtering was accidentally applied only to projects.

### CODE REVIEW

File: model/src/main/kotlin/ScanResult.kt [hunk 1]
```diff
@@ -57,6 +57,8 @@ data class ScanResult(
      * for [provenance]. Findings which [RootLicenseMatcher] assigns as root license files for [path] are also kept.
      */
     fun filterPath(path: String): ScanResult {
+        if (path.isBlank()) return this
```
Comment (sschuberth) [inline model/src/main/kotlin/ScanResult.kt:60]: When you say "In order to make case distinctions (for performance) in the caller unnecessary" I would have expected to also see that code you refer to also being removed as part of this commit. Or do you mean to say that this code now does not need to be *added* in the caller? If so, would you mind clarifying that in the commit message? Also, how is it performance relevant *where* this check is done?

File: scanner/src/main/kotlin/LocalScanner.kt [hunk 2]
```diff
@@ -352,7 +352,9 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
 
         archiveFiles(downloadResult.downloadDirectory, pkg.id, provenance)
 
-        val scanResult = scanPathInternal(downloadResult.downloadDirectory, resultsFile).copy(provenance = provenance)
+        val scanResult = scanPathInternal(downloadResult.downloadDirectory, resultsFile).let { scanResult ->
+            provenance.vcsInfo?.let { scanResult.filterPath(it.path) } ?: scanResult
+        }.copy(provenance = provenance)
```
Comment (sschuberth) [inline scanner/src/main/kotlin/LocalScanner.kt:362]: Commit message nit: As you're referring to the commits as footnotes (as opposed to running text) with plenty of space, would you mind using the full SHA1s?

File: scanner/src/main/kotlin/LocalScanner.kt [hunk 3]
```diff
@@ -215,7 +215,12 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                             "and $scannerDetails, not scanning the package again $packageIndex."
                 }
 
-                storedResults
+                // Due to the bug (fixed in the preceeding commit) the scan results for packages were not properly
```
Comment (sschuberth) [inline scanner/src/main/kotlin/LocalScanner.kt:218]: I believe code comments should not refer to Git commit history in this way, as such references are bound to break. I'd just say "Due to a temporary bug that has been fixed by now ..."

Comment (sschuberth) [reply scanner/src/main/kotlin/LocalScanner.kt:218]: Commit message: s/non applicable/non-applicable/

**Review by sschuberth — DISMISSED** [review]

File: model/src/main/kotlin/ScanResult.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/ScanResult.kt:60]: > Also, how is it performance relevant where this check is done? 

I didn't mean to say that the location of the check is performance relevant.

Comment (fviernau) [reply model/src/main/kotlin/ScanResult.kt:60]: I adjusted the commit message for clarifications.

File: scanner/src/main/kotlin/LocalScanner.kt (hunk 2 above)
Comment (fviernau) [reply scanner/src/main/kotlin/LocalScanner.kt:362]: Done.

File: scanner/src/main/kotlin/LocalScanner.kt (hunk 3 above)
Comment (fviernau) [reply scanner/src/main/kotlin/LocalScanner.kt:218]: Done.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: scanner/src/main/kotlin/LocalScanner.kt [hunk 4]
```diff
@@ -215,7 +215,12 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                             "and $scannerDetails, not scanning the package again $packageIndex."
                 }
 
-                storedResults
+                // Due to a temporary bug that has been fixed by now the scan results for packages were not properly
+                // filtered. Filter them again to fix the problem also scan cache entries which exhibit that problem.
```
[the remark this answers is not in the document]
Comment (fviernau) [reply scanner/src/main/kotlin/LocalScanner.kt:219]: Done.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by fviernau** [comment]
Merged regardless of the non-related Travis failure.

**Pull request merged**